### PR TITLE
v0.5.0: createKeySelector baseFn doesn't receive state

### PR DESCRIPTION
**Pull request by josepot**
@voliva this is the change that we discussed before.

With this, `createKeySelector` will actually enhance the function that it receives :smile:

I need help with the typings, though. Could you please help with that?

### CODE REVIEW

**Comment by voliva** [comment]
Sure thing - I just pushed a commit to this PR

File: types/index.d.ts [hunk 1]
```diff
@@ -38,11 +38,7 @@ export function isInstanceSelector<S, P, R, C>(selector: OutputParametricSelecto
 /// createKeySelector ///
 /////////////////////////
 
-export interface CreateKeySelector {
-  <S>(keySelector: Selector<S, string>): InstanceSelector<S, string>;
-  <S, P>(keySelector: ParametricSelector<S, P, string>): ParametricInstanceSelector<S, P, string>;
-}
-export const createKeySelector: CreateKeySelector;
+export function createKeySelector<P>(keySelector: (props: P) => string): ParametricInstanceSelector<{}, P, string>;
```
Comment (josepot) [inline types/index.d.ts]: Wouldn't it be more accurate like this?
```js
export function createKeySelector<P>(keySelector: (props: P, ...args: any[]) => string): ParametricInstanceSelector<{}, P, string>;
```

**Review by josepot — COMMENTED** [review]
Thanks a lot for helping with the typings @voliva , but I think that you forgot to update the typings of `createKeyedSelectorFactory`?

**Comment by voliva** [comment]
Woops, you're absolutely right - Updated it

**Comment by josepot** [comment]
Awesome! Thanks a tone!

**Pull request merged**